src/category_theory/limits/shapes/images.lean

```diff
@@ -253,6 +253,11 @@ begin
      ... = 𝟙 (image f) ≫ h : by rw [←category.assoc, t]
      ... = h                : by rw [category.id_comp]
 end⟩
+
+lemma epi_of_image_is_iso {X Y : C} (f : X ⟶ Y) [has_image f] [is_iso (image.ι f)]
+  [Π {Z : C} (g h : image f ⟶ Z), has_limit.{v} (parallel_pair g h)] : epi f :=
```

Comment:
This is perhaps an unnecessarily restrictive hypothesis. It is satisfied by typeclass search if `[has_equalizers C]` is available, and I'm inclined to just drop back to that immediately; it seems implausible someone would want to provide the current hypothesis without just going through that.

Reply:
To be honest, I never really understood why you would want either of those. The property you care about is `epi (factor_thru_image f)`, so why not just make that the typeclass argument to `epi_of_image_is_iso`? If the required equalizers exist, then typeclass resolution should have no difficulty concluding `epi (factor_thru_image f)` from that.

Reply:
Maybe provide the current version as a primed version (and the `[]` argument could even be explicit, maybe?).
Then the unprimed version of the lemma could assume `has_equalizers C`.

Reply:
@TwoFX has is right. I've changed this to
```
lemma epi_of_image_is_iso {X Y : C} (f : X ⟶ Y) [has_image f]
  [is_iso (image.ι f)] [epi (factor_thru_image f)] : epi f := ...
```

Reply:
(There's no value in keeping the current version with a prime, because if you have either `has_equalizers C`, or the pi type, then the new version fires in any case.)

Reply:
(Reid may pop by to note what a bad idea it is to prove things using typeclass search, and I won't disagree. We should probably make a bunch of things not typeclasses.)